core/src/main/java/com/linecorp/armeria/common/AbstractHttpMessageBuilder.java

```diff
@@ -0,0 +1,148 @@
+                                   requireNonNull(value, "value"));
+        return this;
+    }
+
+    AbstractHttpMessageBuilder headers(
+            Iterable<? extends Entry<? extends CharSequence, String>> headers) {
+        requireNonNull(headers, "headers");
+        headersBuilder().add(headers);
+        return this;
+    }
+
+    AbstractHttpMessageBuilder content(String content) {
+        return content(MediaType.PLAIN_TEXT_UTF_8, content);
+    }
+
+    AbstractHttpMessageBuilder content(MediaType contentType, CharSequence content) {
+        requireNonNull(contentType, "contentType");
+        return content(contentType,
+                       HttpData.of(contentType.charset(StandardCharsets.UTF_8),
+                                   content));
+    }
+
+    AbstractHttpMessageBuilder content(MediaType contentType, String content) {
+        requireNonNull(contentType, "contentType");
+        return content(contentType, HttpData.of(contentType.charset(StandardCharsets.UTF_8),
+                                                content));
+    }
+
+    @FormatMethod
+    AbstractHttpMessageBuilder content(@FormatString String format, Object... content) {
+        return content(MediaType.PLAIN_TEXT_UTF_8, format, content);
+    }
+
+    @FormatMethod
+    AbstractHttpMessageBuilder content(MediaType contentType, @FormatString String format,
+                                       Object... content) {
+        requireNonNull(contentType, "contentType");
+        return content(contentType, HttpData.of(contentType.charset(StandardCharsets.UTF_8),
+                                                format, content));
+    }
+
+    AbstractHttpMessageBuilder content(MediaType contentType, byte[] content) {
+        return content(contentType, HttpData.wrap(content));
+    }
+
+    AbstractHttpMessageBuilder content(MediaType contentType, HttpData content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+        checkState(publisher == null, "publisher has been set already");
+        headersBuilder().contentType(contentType);
+        this.content = content;
+        return this;
+    }
+
+    AbstractHttpMessageBuilder content(MediaType contentType, Publisher<? extends HttpData> content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "publisher");
+        checkState(this.content == null, "content has been set already");
+        headersBuilder().contentType(contentType);
+        publisher = content;
+        return this;
+    }
+
+    AbstractHttpMessageBuilder contentJson(Object content) {
+        requireNonNull(content, "content");
+        headersBuilder().contentType(MediaType.JSON);
+        try {
+            this.content = HttpData.wrap(JacksonUtil.writeValueAsBytes(content));
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e.toString(), e);
```

Comment:
nit:
```suggestion
            throw new IllegalArgumentException("failed to serialize " + content, e);
```
Let's fix it here as well. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/HttpResponse.java#L558